service/frontend/service.go

```diff
@@ -254,9 +254,7 @@ func (s *Service) Start() {
 	// Additional decorations
 	var handler Handler = s.handler
 	handler = NewDCRedirectionHandler(handler, s, s.config, s.params.DCRedirectionPolicy)
-	if s.params.Authorizer != nil {
```

Comment:
The reason we remove this check is to allow creating Authorizer within the service, so that we can have the `domainCache` entity. 
We have to allow boostraping(server.go) passing in a nil as Authorizer, and then create an Authorizer based on the config. 
Uber's internal repo doing the customized boostraping which will pass in a non-nil or nil Authorizer will not have any behavior change. By default this AccessControllerHandler is noop.